manim/mobject/types/vectorized_mobject.py

```diff
@@ -1096,6 +1260,134 @@ def get_subpaths(self) -> tuple:
         """
         return self.get_subpaths_from_points(self.points)
 
+    def get_subpath_split_indices_from_points(
+        self, points: np.ndarray, n_dims: int = 3, strip_null_end_curves: bool = False
+    ) -> np.ndarray:
+        nppcc = self.n_points_per_cubic_curve
+        starts = points[::nppcc]
+        ends = points[nppcc - 1 :: nppcc]
+        # This ensures that there are no more starts than ends.
+        # TODO: ends.shape[0] would be more efficient, but some test cases regarding
+        # Flash and ShowPassing flash expect a Python list instead of an ndarray, so
+        # ends.shape[0] breaks those test cases. Fix these inconsistencies.
+        n_curves = len(ends)
+        starts = starts[:n_curves]
+
+        # DO NOT DELETE THIS! Or else, a [[0 0]] ndarray will be returned
+        # for an empty list of points, when it should instead return [].
+        if n_curves == 0:
+            return np.empty((0, 2), dtype=int)
+        if n_curves == 1:
+            return np.array([[0, nppcc]])
+
+        if n_dims == 2:
+            is_equal = self.consider_points_equals_2d
+            is_different = self.consider_points_different_2d
+        else:
+            is_equal = self.consider_points_equals
+            is_different = self.consider_points_different
+
+        is_far = is_different(starts[1:], ends[:-1])
```

Comment:
Semantics: I would have expected `is_far` to be a function (just like `is_different`, `is_equal`), rather than an array -- it might be worth renaming the variable.

Reply:
Good point. That also bothered me a bit, but I didn't know which names could be better. I'll take a deeper look at this issue.

Reply:
Fixed! The new names should no longer cause confusion IMO.